### ISSUE TITLE
proposal: have checksums of files installed by kiss from package to find changed/broken files

### ISSUE THREAD

**Issue by sewnie** [kiss-community/repo#98]
related: `xbps-pkgdb -a`, this utility uses the package manager's database of file checksums installed by a package and checks if they match, i used to find this quite useful on Void to easily find broken files or configuration files.

personally, i never found a use for this when i switched to kiss linux.

**Comment by aabacchus** [comment]
This already happens for files in /etc, are you suggesting to checksum every file installed?

**Comment by sewnie** [comment]
Yes

**Comment by illiliti** [comment]
We would have to switch to a more faster hash algo because sha256 is slow.

**Comment by aabacchus** [comment]
Do you often find files to have been accidentally edited? Even if you do, storing checksums of every file can only tell you if a file has been changed, and can't help in restoring it. For some packages, this store would take a long time to make and be relatively large in size.

The existing manifest can be used to make sure no files have been deleted.

**Comment by sewnie** [comment]
> and can't help in restoring it. 

yes, I at least know what package the file belongs to and I can reinstall it.

**Comment by git-bruh** [comment]
https://codeberg.org/kiss-community/repo/issues/98
